### PR TITLE
!feat: ai sdk and configuration

### PR DESCRIPTION
**Pull request by okxiaoliang4**
Add `@ai/sdk` for multiple ai provider including ollama

closed: #27 #22 #42

- [x] ai package
- [x] ollama
- [x] azure
- [x] openai
- [x] anthropic

### CODE REVIEW

**Comment by okxiaoliang4** [comment]
Should i merge into this https://github.com/okxiaoliang4/void/pull/1 together? Will improve the setting configuration, can closed #42

**Comment by okxiaoliang4** [comment]
Fully configuration.

```json
{
  "void.provider": "azure",
  "void.anthropic.apiKey": "",
  "void.anthropic.model": "claude-3-5-sonnet-20240620",
  "void.anthropic.providerSettings": {
    "baseURL": "",
    "headers": {}
  },
  "void.openai.apiKey": "",
  "void.openai.model": "gpt-4o",
  "void.openai.providerSettings": {
    "baseURL": "",
    "compatibility": "strict",
    "headers": {},
    "organization": "",
    "project": ""
  },
  "void.ollama.model": "llama3.1",
  "void.ollama.providerSettings": {
    "baseURL": "",
    "headers": {}
  },
  "void.azure.apiKey": "xxxx",
  "void.azure.deploymentId": "gpt-4o",
  "void.azure.resourceName": "xxxx",
  "void.azure.providerSettings": {
    "baseURL": "",
    "headers": {}
  },
  "void.greptile.apiKey": "apik_xxxx",
  "void.greptile.providerSettings": {
    "headers": {
      "X-Github-Token": "xxxx"
    },
    "repoinfo": [
      {
        "branch": "main",
        "remote": "github",
        "repository": "xxxx"
      }
    ]
  },
}
```

breaking change:

void.whichApi -> void.provider
void.anthropicApiKey -> void.anthropic.apiKey
void.openAIApiKey -> void.openai.apiKey
void.greptileApiKey -> void.greptile.apiKey
void.githubPAT -> void.greptile.providerSetting.headers['X-Github-Token']
void.ollamaSettings -> void.ollama.*

**Comment by mathewpareles** [comment]
Yes, please merge these changes.

**Comment by okxiaoliang4** [comment]
> Yes, please merge these changes

Is merged. This PR is ready to review and merge

File: extensions/void/package.json [hunk 1]
```diff
@@ -15,35 +15,274 @@
...
+          "description": "Greptile API key."
+        },
+        "void.greptile.providerSettings": {
+          "type": "object",
+          "properties": {
+            "headers": {
+              "type": "object",
+              "description": "Custom headers to include in the requests.",
+              "properties": {
+                "X-Github-Token": {
```
Comment (okxiaoliang4) [inline extensions/void/package.json:259]: Should we change this to `void.greptile.githubToken`?

Comment (andrewpareles) [reply extensions/void/package.json:259]: I think so, yes.

**Comment by andrewpareles** [comment]
Unfortunately we don't want to have a dependence on 'ai'. Please see [`CONTRIBUTING.md`](https://github.com/voideditor/void/blob/main/CONTRIBUTING.md). Thanks for your contribution, and we'll add in azure support soon.

**Pull request closed without merging**